core/src/main/java/com/datastax/oss/driver/internal/core/cql/Conversions.java

```diff
@@ -489,32 +488,27 @@ static int sizeOfSimpleStatementValues(
       ProtocolVersion protocolVersion,
       CodecRegistry codecRegistry) {
     int size = 0;
+
     if (!simpleStatement.getPositionalValues().isEmpty()) {
-      int totalValuesSize =
-          Values.sizeOfPositionalValues(
-              simpleStatement
-                  .getPositionalValues()
-                  .stream()
-                  // need to convert Object into ByteBuffer
-                  .map(val -> Conversions.encode(val, codecRegistry, protocolVersion))
-                  .collect(Collectors.toList()));
-      size += totalValuesSize;
+
+      List<ByteBuffer> positionalValues =
```

Comment:
Instead of building up a new List (or Map in the named value case), you could create a running count in place, using `Values.sizeOfValue` for each value and then adding `PrimtiveSizes.SHORT` at the end for the positional value case . That would prevent need for building a new collection, but on the other hand it requires this code to have knowledge about how value collections are sized, so maybe it's better for this to be the done the way you've currently done it as i don't think this code needs to be particularly performant since its not on the hot path (user has to call `computeSizeInBytes`)